esphome/components/speed/fan/__init__.py

```diff
@@ -40,3 +42,6 @@ async def to_code(config):
     if CONF_DIRECTION_OUTPUT in config:
         direction_output = await cg.get_variable(config[CONF_DIRECTION_OUTPUT])
         cg.add(var.set_direction(direction_output))
+
+    if CONF_OFF_SPEED_CYCLE in config:
+        cg.add(var.set_offcycle(config[CONF_OFF_SPEED_CYCLE]))
```

Comment:
Since you added a default, it will always be in config
```suggestion
   cg.add(var.set_offcycle(config[CONF_OFF_SPEED_CYCLE]))
```

Reply:
Resolved by b1ebc789db86ae2c1d2792cf3d48621c82a42f49